app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -186,10 +206,28 @@ fun ActivityDetailsScreen(
   val nbActivitiesCreated = activitiesList.filter { it.creator == creator.id }.size
   val hourDateViewModel = HourDateViewModel()
 
+  // Fetch participants and update list
+  var participantsList by remember { mutableStateOf(listOf<User>()) }
+  LaunchedEffect(activity?.uid) {
+    activity?.participants?.forEach { participant ->
+      if (participant.id.isEmpty()) participantsList = participantsList + participant
+      else {
+        if (participant.id.length > 8) {
+          profileViewModel.getUserData(
+              participant.id,
+              onResult = {
+                if (it != null && !participantsList.contains(it))
+                    participantsList = participantsList.plus(it)
+              })
+        }
+      }
+    }
+  }
```

Comment:
this will permit updating the list of participants in case of updates in the database